lib/xnvme_spec.c

```diff
@@ -43,7 +43,7 @@ xnvme_spec_log_health_fpr(FILE *stream, const struct xnvme_spec_log_health_entry
 
 	wrtn += fprintf(stream, "\n");
 	wrtn += fprintf(stream, "  crit_warn: %" PRIu8 "\n", log->crit_warn);
-	wrtn += fprintf(stream, "  comp_temp: %" PRIu16 "\n", log->comp_temp - 273u);
+	wrtn += fprintf(stream, "  comp_temp: %" PRIu16 "\n", (uint16_t)(log->comp_temp - 273u));
```

Comment:
By subtracting 273 you are converting from K to C. The result may very well be negative. Just let the compiler promote `comp_temp` to `int`.

Reply:
Right, thanks! Fixed.